google-cloud-bigtable/lib/google/cloud/bigtable/row_filter/chain_filter.rb

```diff
@@ -529,16 +529,16 @@ def column_range range
           #  filter.length # 2
           #
           def length
-            @grpc.filters.length
+            @filters.length
           end
 
           ##
-          # Gets the list of filters.
+          # Returns the frozen filters array.
           #
-          # @return [Array<Google::Bigtable::V2::RowFilter>]
+          # @return [Array<SimpleFilter|ChainFilter|InterleaveFilter|ConditionFilter>]
           #
           def filters
-            @grpc.filters
+            @filters.freeze
```

Comment:
This will prevent further filters from being added. Is that what we want? (Or should we dup the array?)

Reply:
To be honest, I have no idea why the underlying protobuf `filters` list was being exposed. This method is not called anywhere in the project that I can see. I am open to any suggestions regarding this method, including removing it completely.

Reply:
> This will prevent further filters from being added. Is that what we want?

Yes, making additions to the list of filters should be done via the provided methods on this class, such as `condition`, `qualifier`, etc.

Reply:
I can imagine that an application would want to get the elements of a chain. (Maybe it's displaying a UI.) I'm just saying that the act of getting that data shouldn't cause the underlying array to freeze (thus preventing further additions).

Reply:
Yikes, of course. Good catch. I'll dup the array.